// shopify/dist/node-package-manager-IEZ6F5VN.js
import {
  FindUpAndReadPackageJsonNotFoundError,
  PackageJsonNotFoundError,
  UnknownPackageManagerError,
  addNPMDependencies,
  addNPMDependenciesIfNeeded,
  addNPMDependenciesWithoutVersionIfNeeded,
  addResolutionOrOverride,
  bunLockfile,
  checkForCachedNewVersion,
  checkForNewVersion,
  findUpAndReadPackageJson,
  getDependencies,
  getPackageManager,
  getPackageName,
  getPackageVersion,
  inferPackageManager,
  installNPMDependenciesRecursively,
  installNodeModules,
  lockfiles,
  npmLockfile,
  packageManager,
  packageManagerFromUserAgent,
  pnpmLockfile,
  pnpmWorkspaceFile,
  readAndParsePackageJson,
  usesWorkspaces,
  versionSatisfies,
  writePackageJSON,
  yarnLockfile
} from "./chunk-FDJVVEGH.js";
import "./chunk-BYPQXSAL.js";
import "./chunk-PJKCWDDX.js";
import "./chunk-XVNW332R.js";
import "./chunk-XEDHKQZA.js";
import "./chunk-T2GATXSP.js";
import "./chunk-CBD6UXLA.js";
import "./chunk-CBBS4CV7.js";
import "./chunk-SNOECVP4.js";
import "./chunk-NEOIQ6PN.js";
import "./chunk-7Q3MMWAC.js";
import "./chunk-MX4RJDBM.js";
import "./chunk-NB4NLOEJ.js";
import "./chunk-UBB7JKND.js";
import "./chunk-522OB3EU.js";
import "./chunk-PNN7RS7Y.js";
import "./chunk-POZ5MGPT.js";
export {
  FindUpAndReadPackageJsonNotFoundError,
  PackageJsonNotFoundError,
  UnknownPackageManagerError,
  addNPMDependencies,
  addNPMDependenciesIfNeeded,
  addNPMDependenciesWithoutVersionIfNeeded,
  addResolutionOrOverride,
  bunLockfile,
  checkForCachedNewVersion,
  checkForNewVersion,
  findUpAndReadPackageJson,
  getDependencies,
  getPackageManager,
  getPackageName,
  getPackageVersion,
  inferPackageManager,
  installNPMDependenciesRecursively,
  installNodeModules,
  lockfiles,
  npmLockfile,
  packageManager,
  packageManagerFromUserAgent,
  pnpmLockfile,
  pnpmWorkspaceFile,
  readAndParsePackageJson,
  usesWorkspaces,
  versionSatisfies,
  writePackageJSON,
  yarnLockfile
};
